src/group_theory/double_coset.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2021 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+import data.setoid.basic
+import group_theory.subgroup.basic
+import group_theory.coset
+import group_theory.subgroup.pointwise
+import data.set.basic
+import tactic.group
+
+/-!
+# Double cosets
+
+This file defines double cosets for two subgroups `H K` of a group `G` and the quotient of `G` by
+the double coset relation, i.e. `H \ G / K`. We also prove that `G` can be writen as a disjoint
+union of the double cosets and that if one of `H` or `K` is the trivial group (i.e. `⊥` ) then
+this is the usual left or right quotient of a group by a subgroup.
+
+## Main definitions
+
+* `double_coset_rel`: The double coset relation defined by two subgroups `H K` of `G`.
+* `double_coset.quotient`: The quotient of `G` by the double coset relation, i.e, ``H \ G / K`.
+-/
+
+
+variables {G : Type*} [group G] {α : Type*} [has_mul α] (J: subgroup G) (g : G)
+
+namespace double_coset
+
+open_locale pointwise
+
+/--The double_coset as an element of `set α` corresponding to `s a t` -/
+def doset (a : α) (s t : set α) : set α := s * {a} * t
```

Comment:
Do you want this to be called `double_coset.doset`? Or should it be in the root namespace?

Reply:
I have no preference really. I'm happy to go with what you think is best.

Reply:
`double_coset.doset` seems more duplication then needed. So I suggest `doset` in the root namespace.
The entire `double_coset` namespace can then be replaced by the `doset` namespace. This might also allow you to make some names shorter in the rest of the PR.

Reply:
Sure thing!